v2/fb-xcom-api-validator/tests/capabilities.test.ts

```diff
@@ -1,77 +1,72 @@
 
     beforeAll(async () => {
-      const client = new Client();
-      result = (await client.capabilities.getCapabilities({})) as Capabilities;
+      capabilitiesResponse = (await client.capabilities.getCapabilities({})) as Capabilities;
     });
 
-    it('should include a valid api version', () => {
-      expect(isValidApiVersion(result.version)).toBe(true);
+    it('should match config capabilities', () => {
+      expect(capabilitiesResponse).toEqual(capabilitiesConfig);
     });
 
-    describe('Component', () => {
-      describe.each(requiredCapabilitiesComponents)('%s (required)', (componentName) => {
-        it('should be listed in response', () => {
-          expect(result.components[componentName]).toBeDefined();
-        });
-
-        it('should have a valid value', () => {
-          expect(isValidComponentValue(result.components[componentName])).toBe(true);
-        });
-      });
-
-      describe.each(optionalCapabilitiesComponents)('%s (optional)', (componentName) => {
-        it('should have a valid value when listed in the response', () => {
-          expect(isValidComponentValue(result.components[componentName])).toBe(true);
-        });
-      });
+    it('should include a known api version', () => {
+      expect(isKnownApiVersion(capabilitiesConfig.version)).toBe(true);
     });
   });
-});
 
-function isValidApiVersion(value: string) {
-  const apiVersionRegex = /^\d+\.\d+\.\d+$/;
-  return apiVersionRegex.test(value);
-}
+  describe('/capabilities/assets', () => {
+    let result: { assets: AssetDefinition[] };
 
-function isStringArray(value: unknown) {
-  if (!Array.isArray(value)) {
-    return false;
-  }
-
-  if (value.find((item) => typeof item !== 'string')) {
-    return false;
-  }
+    beforeAll(async () => {
+      result = (await client.capabilities.getAdditionalAssets({})) as { assets: AssetDefinition[] };
+    });
 
-  return true;
-}
+    it('should return list of assets', () => {
+      expect(result.assets).toBeDefined();
+    });
 
-function isValidComponentValue(value: unknown) {
-  if (value === undefined) {
-    return true;
-  }
+    describe('Interaction with /capabilities/assets/:id', () => {
+      const isListedAsset = async (assetId: string): Promise<boolean> => {
+        try {
+          const asset = (await client.capabilities.getAssetDetails({
+            id: assetId,
+          })) as AssetDefinition;
+          if (!asset || asset.id !== assetId) {
+            return false;
+          }
+          return true;
+        } catch (err) {
+          if ((err as any).body?.errorType === ErrorType.NOT_FOUND) {
+            return false;
+          }
+          throw err;
+        }
+      };
 
-  if (value === '*' || isStringArray(value)) {
-    return true;
-  }
+      it('should find each listed asset on asset details endpoint', async () => {
+        for (const asset of result.assets) {
+          const found = await isListedAsset(asset.id);
+          expect(found).toBe(true);
```

Comment:
When this condition fails, we won't know on which asset it fails. That's why getting all the data, and pass it to a predicate is better. Or you could use `fail` from jest-extended.